app/providers/kubernetes/list.go

```diff
@@ -3,33 +3,51 @@ package kubernetes
 import (
 	"context"
 	"github.com/sablierapp/sablier/app/discovery"
-	"github.com/sablierapp/sablier/app/providers"
 	"github.com/sablierapp/sablier/app/types"
 	log "github.com/sirupsen/logrus"
 	v1 "k8s.io/api/apps/v1"
 	core_v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
 	"strconv"
-	"strings"
 )
 
-func (provider *KubernetesProvider) InstanceList(ctx context.Context, options providers.InstanceListOptions) ([]types.Instance, error) {
-	deployments, err := provider.deploymentList(ctx, options)
+const (
+	LabelEnable                      = "sablierapp.dev/enable"
+	LabelGroup                       = "sablierapp.dev/group"
+	LabelGroupDefaultValue           = "default"
+	LabelReplicas                    = "sablierapp.dev/replicas"
+	LabelReplicasDefaultValue uint64 = 1
+)
+
+func (provider *KubernetesProvider) List(ctx context.Context) ([]types.Instance, error) {
+	deployments, err := provider.deploymentList(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	statefulSets, err := provider.statefulSetList(ctx, options)
+	statefulSets, err := provider.statefulSetList(ctx)
 	if err != nil {
 		return nil, err
 	}
 
 	return append(deployments, statefulSets...), nil
 }
 
-func (provider *KubernetesProvider) deploymentList(ctx context.Context, options providers.InstanceListOptions) ([]types.Instance, error) {
+func (provider *KubernetesProvider) deploymentList(ctx context.Context) ([]types.Instance, error) {
+	requirement, err := labels.NewRequirement(LabelEnable, selection.Equals, []string{"true"})
+	if err != nil {
+		return nil, err
+	}
+	requirementDeprecated, err := labels.NewRequirement(discovery.LabelEnable, selection.Equals, []string{"true"})
+	if err != nil {
+		return nil, err
+	}
+	selector := labels.NewSelector()
+	selector = selector.Add(*requirement, *requirementDeprecated)
```

Comment:
If you merge the 2 selector type here, the deployments will need to have both legacy and new labels because the selector would result in `sablier.enable=true,sablierapp.dev/enable=true`.

I don't know how to handle this, either deprecate old selectors or add a flag to toggle legacy/new (or make it customizable?)

Reply:
After a bit more tweaking I got it to work, I added a bunch of Trace logging to help me find the root cause and it was mainly due to old `discovery.LabelGroup` usage.

I rebased main onto my branch (449-xxx) so I cannot open a PR on yours but you can see my diff here: https://github.com/sablierapp/sablier/compare/main...BapRx:sablier:449-kubernetes-labels-not-public

Reply:
Awesome, I will check it out.